src/main/kotlin/io/curity/dynamoDBDataAccessProvider/DynamoDBClient.kt

```diff
@@ -51,51 +51,100 @@ import software.amazon.awssdk.services.sts.StsClient
 import software.amazon.awssdk.services.sts.model.AssumeRoleRequest
 import software.amazon.awssdk.services.sts.model.AssumeRoleResponse
 import software.amazon.awssdk.services.sts.model.Credentials
+import java.net.URI
+import java.util.*
 
-class DynamoDBClient(private val config: DynamoDBDataAccessProviderDataAccessProviderConfig): ManagedObject<DynamoDBDataAccessProviderDataAccessProviderConfig>(config)
-{
+class DynamoDBClient(private val config: DynamoDBDataAccessProviderDataAccessProviderConfig) : ManagedObject<DynamoDBDataAccessProviderDataAccessProviderConfig>(config) {
     private val _awsRegion = Region.of(config.getAwsRegion().awsRegion)
-    private val client: DynamoDbClient = initializeDynamoDBClient()
+    private val client: Optional<DynamoDbClient> = initializeDynamoDBClient()
 
-    private fun initializeDynamoDBClient(): DynamoDbClient
-    {
+    private fun initializeDynamoDBClient(): Optional<DynamoDbClient> {
         val accessMethod = config.getDynamodbAccessMethod()
         /*Use Instance Profile from IAM Role applied to EC2 instance*/
-        var creds: AwsCredentialsProvider?
+        var creds: Optional<AwsCredentialsProvider>
 
-        if (accessMethod.isEC2InstanceProfile.isPresent && accessMethod.isEC2InstanceProfile.get()) {
```

Comment:
This line had the bug ```accessMethod.isEC2InstanceProfile.get()``` would trigger a crash if any other configuration method was used other than EC2InstanceProfile.